packages/postgres/src/pool.ts

```diff
@@ -470,19 +470,39 @@ export class PostgresPool {
    * will be committed otherwise.
    */
   async runInTransactionAsync<T>(fn: (client: pg.PoolClient) => Promise<T>): Promise<T> {
-    const client = await this.beginTransactionAsync();
+    // Check if we're already inside a transaction. If so, we won't start another one,
+    // as Postgres doesn't support nested transactions.
+    let client = this.alsClient.getStore();
+    const isNestedTransaction = client !== undefined;
+    if (!client) {
+      client = await this.beginTransactionAsync();
+    }
+
+    // We reassign to a separate variable here so that TypeScript can reason
+    // about the fact that `client` will always be defined by this point.
+    const transactionClient = client;
```

Comment:
```suggestion
    const transactionClient = client || await this.beginTransactionAsync();
```

This is nit, of course, but I think it makes the code slightly more readable. Also, this allows `client` to be a `const`.

Reply:
That is indeed much nicer - thanks! I'll tweak it to use `??` instead of `||`.